src/server/scripts/Commands/cs_lookup.cpp

```diff
@@ -1621,19 +1633,20 @@ class lookup_commandscript : public CommandScript
         stmt->SetData(0, email);
         PreparedQueryResult result = LoginDatabase.Query(stmt);
 
+        if (!result)
+        {
+            handler->SendErrorMessage(LANG_EMAIL_NOT_FOUND, email);
+            return false;
+        }
+
         return LookupPlayerSearchCommand(result, *limit ? *limit : -1, handler);
     }
 
     static bool LookupPlayerSearchCommand(PreparedQueryResult result, int32 limit, ChatHandler* handler)
     {
-        if (!result)
```

Comment:
Please explain this one. If it didn't find any result the loop at 1652 is void entirely, so the early return made sense

Reply:
It doesn't really answer my question, nothing in this function will execute if the result is null, except the two lines you added at the bottom. So move the zero result lines here or keep the early return as is as as I see it already says no players were found anyway